src/librustc/infer/error_reporting/nice_region_error/mod.rs

```diff
@@ -65,10 +65,16 @@ impl<'cx, 'gcx, 'tcx> NiceRegionError<'cx, 'gcx, 'tcx> {
     }
 
     pub fn try_report(&self) -> Option<ErrorReported> {
```

Comment:
I think we should add a `try_report_from_nll` method instead, and invoke that from the NLL borrow checker, vs checking the flag. Just seems a bit cleaner, but it may also improve the error messages in some cases below